accounts/abi/abi.go

```diff
@@ -149,14 +150,36 @@ func (abi ABI) Call(executer Executer, name string, args ...interface{}) interfa
 }
 
 func (abi *ABI) UnmarshalJSON(data []byte) error {
-	var methods []Method
-	if err := json.Unmarshal(data, &methods); err != nil {
+	var fields []struct {
+		Type    string
+		Name    string
+		Const   bool
+		Indexed bool
+		Inputs  []Argument
+		Outputs []Argument
+	}
+
+	if err := json.Unmarshal(data, &fields); err != nil {
 		return err
 	}
 
 	abi.Methods = make(map[string]Method)
-	for _, method := range methods {
-		abi.Methods[method.Name] = method
+	abi.Events = make(map[string]Event)
+	for _, field := range fields {
+		switch field.Type {
+		case "function":
```

Comment:
According to https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI#json, we can have a `constructor` field too and they might also be fully omitted, defaulting to `function`.